File: Master/Year 1/Semester 1/AMSS/Proiect/app/src/main/java/ro/robertto/rescuepets/presentation/activities/ConfirmEmailActivity.java
package ro.robertto.rescuepets.presentation.activities;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

import ro.robertto.rescuepets.R;

public class ConfirmEmailActivity extends AppCompatActivity {

    private final String gmailPackageName = "com.google.android.gm";
    private final String playStorePackageName = "com.android.vending";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_confirm_email);

        Button gmailButton = findViewById(R.id.gmail_button);
        gmailButton.setOnClickListener(view -> {
            Intent gmailIntent = getPackageManager().getLaunchIntentForPackage(gmailPackageName);
            if (gmailIntent != null) {
                startActivity(gmailIntent);
            } else {
                Intent playStoreIntent = new Intent(Intent.ACTION_VIEW);
                playStoreIntent.setData(Uri.parse("market://details?id=" + gmailPackageName));
                playStoreIntent.setPackage(playStorePackageName);
                startActivity(playStoreIntent);
            }
        });

        Button outlookButton = findViewById(R.id.outlook_button);
        outlookButton.setOnClickListener(view -> {
            String outlookPackageName = "com.microsoft.office.outlook";
            Intent outlookIntent = getPackageManager().getLaunchIntentForPackage(outlookPackageName);
            if (outlookIntent != null) {
                startActivity(outlookIntent);
            } else {
                Intent playStoreIntent = new Intent(Intent.ACTION_VIEW);
                playStoreIntent.setData(Uri.parse("market://details?id=" + outlookPackageName));
                playStoreIntent.setPackage(playStorePackageName);
                startActivity(playStoreIntent);
            }
        });

        Button loginButton = findViewById(R.id.login_button);
        loginButton.setOnClickListener(view -> {
            startActivity(new Intent(this, LogInActivity.class));
        });
    }
}

File: Anul 3/Semestrul 2/Android/Proiect/EmergencyApp/app/src/main/java/com/lifeSavers/emergencyapp/activities/ConfirmEmailActivity.kt
package com.lifeSavers.emergencyapp.activities

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.lifeSavers.emergencyapp.R

class ConfirmEmailActivity : AppCompatActivity() {

    private val gmailPackageName = "com.google.android.gm"
    private val playStorePackageName = "com.android.vending"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_confirm_email)

        val gmailButton = findViewById<Button>(R.id.gmail_button)
        gmailButton.setOnClickListener {
            val gmailIntent = packageManager.getLaunchIntentForPackage(gmailPackageName)
            if (gmailIntent != null) {
                startActivity(gmailIntent)
            } else {
                val playStoreIntent = Intent(Intent.ACTION_VIEW).apply {
                    data = Uri.parse("market://details?id=$gmailPackageName")
                    setPackage(playStorePackageName)
                }
                startActivity(playStoreIntent)
            }
        }

        val outlookButton = findViewById<Button>(R.id.outlook_button)
        outlookButton.setOnClickListener {
            val outlookPackageName = "com.microsoft.office.outlook"
            val outlookIntent = packageManager.getLaunchIntentForPackage(outlookPackageName)
            if (outlookIntent != null) {
                startActivity(outlookIntent)
            } else {
                val playStoreIntent = Intent(Intent.ACTION_VIEW).apply {
                    data = Uri.parse("market://details?id=$outlookPackageName")
                    setPackage(playStorePackageName)
                }
                startActivity(playStoreIntent)
            }
        }

        val loginButton = findViewById<Button>(R.id.login_button)
        loginButton.setOnClickListener {
            startActivity(Intent(this, LogInActivity::class.java))
        }
    }
}
